### ISSUE TITLE
A question about CoordConv

### ISSUE THREAD

**Issue by SugarMasuo** [cvlab-stonybrook/doc3D-dataset#8]
Hi, do you replace all tranditional Conv in Texture Mapping Network using CoordConv ? Thanks~

**Comment by SugarMasuo** [comment]
And, what kind of upsample should be used in this network?

**Comment by sagniklp** [comment]
No, it's just used in the first layer of the network. 
We use ConvTranspose2D for upsampling.

**Comment by Enuvesta** [comment]
Also, do you have a ploblem with document borders appearing on feature maps in the texture network?

**Comment by SugarMasuo** [comment]
@sagniklp is the encoder and decoder constructed by dense block directly ? And the first layer of first dense block used CoordConv instead of Conv . Am I right？

**Comment by sagniklp** [comment]
Yes.

**Comment by sagniklp** [comment]
> Also, do you have a ploblem with document borders appearing on feature maps in the texture network?

@Enuvesta Could you show me an example? I'm not sure about your concern.

**Comment by SugarMasuo** [comment]
> @sagniklp is the encoder and decoder constructed by dense block directly ? And the first layer of first dense block used CoordConv instead of Conv . Am I right？

> Yes.

So, there is no the first 7*7 Conv before the first dense block( which is  like structure in the original paper of DenseNet), right? Thanks~

**Comment by Enuvesta** [comment]
@sagniklp Here is an exmaple:
![фчфьзду](https://user-images.githubusercontent.com/10676265/66893769-74613900-eff7-11e9-8163-a70c78fe8e24.png)
Left is predicted bm, middle is GT bm and left is input image

**Comment by sagniklp** [comment]
@Enuvesta  
1. Backward mapping should have a shape [448,448,2] . It seems like these images have 3 channels. Could you detail more about what you're exactly doing?

2. If you're trying to regress backward mapping then input is the wc [448,448,3] and output is a bm [448,448,2]. 

3. GT bm should look like this if you plot each channel.
![Screenshot from 2019-10-16 14-18-08](https://user-images.githubusercontent.com/10586359/66947074-34fb1280-f020-11e9-9515-43e1c0b4b24a.png)

**Comment by SugarMasuo** [comment]
> @Enuvesta
> 
> 1. Backward mapping should have a shape [448,448,2] . It seems like these images have 3 channels. Could you detail more about what you're exactly doing?
> 2. If you're trying to regress backward mapping then input is the wc [448,448,3] and output is a bm [448,448,2].
> 3. GT bm should look like this if you plot each channel.
>    ![Screenshot from 2019-10-16 14-18-08](https://user-images.githubusercontent.com/10586359/66947074-34fb1280-f020-11e9-9515-43e1c0b4b24a.png)

Hi, @sagniklp, I'm confused with the network too, especially the texture mapping network. Is it convenient to share your network structure with us? Thanks!

**Comment by sagniklp** [comment]
We will release our code before ICCV. 
What network are you using now? Our network is encoder-decoder of densenet blocks.

**Comment by SugarMasuo** [comment]
That is worth looking forward to!
I am a green hand at DL. I'm trying to build the network of  encoder-decoder using densenet blocks.

**Comment by Enuvesta** [comment]
I'm using encoder-decoder architecture consisted of 5 up and 5 down densenet blocks.
Waiting for the updates!

**Comment by sagniklp** [comment]
What is the shape of the bottleneck layer? Last layer of the decoder.

Sagnik Das

**Comment by Enuvesta** [comment]
[-1, 2, 448, 448]
Or do you mean encoder? 
I've used an implementation presented here https://github.com/bfortuner/pytorch_tiramisu. More specifically FCDenseNet67 with 2 classes.

**Comment by sagniklp** [comment]
Remove the UNet like skip connections between encoder and decoder blocks. I think that's the reason for the background showing up in the final prediction. 
I hope you don't have the softmax since it's a regression network.

**Comment by Enuvesta** [comment]
Removing skip connections worked! 
Thank you so much!

**Comment by SugarMasuo** [comment]
hello @sagniklp , What are the criteria for dividing images into different folders?  Is it ok if I put all files in different folders into the same folder during training? Thanks

**Comment by SugarMasuo** [comment]
> Removing skip connections worked!
> Thank you so much!

Could you share loss curve of train and val? I'm new at DL, my loss is so big and it converges very slowly. Thanks!

**Comment by sagniklp** [comment]
> hello @sagniklp , What are the criteria for dividing images into different folders? Is it ok if I put all files in different folders into the same folder during training? Thanks

It's fine just make sure train and val contains different meshes. If you split the filename with '-' the first item is the mesh id.

**Comment by SugarMasuo** [comment]
> > hello @sagniklp , What are the criteria for dividing images into different folders? Is it ok if I put all files in different folders into the same folder during training? Thanks
> 
> It's fine just make sure train and val contains different meshes. If you split the filename with '-' the first item is the mesh id.

I see, so for example, 1_1_3-pp_Page_127-5ih0001.png, 1_1_3 is the mesh id, but the number 1, 1, 3 means for what? 
And I find the same mesh id is distributed in different folders, is there some separate folder suitable for val? 
What is the meaning of following words, like pp_Page_127 and 5ih0001? Thanks~

**Comment by sagniklp** [comment]
Names follow this convention:  
meshid-texture_name-random_string 

You can use this split:
[meshsplit.txt](https://github.com/cvlab-stonybrook/doc3D-dataset/files/3774459/meshsplit.txt)
This file should be readable as a `.json`.

**Comment by sagniklp** [comment]
Because the dataset also contains child meshes, . The main use of the split
is to make sure there are no common to parent meshes, otherwise the model
starts to memorize the meshes. Mesh ID can be broken as
childmesh_parentmeshA_parentmeshB. Each mesh (parentmeshA) is first flipped
and 8 meshes are produced with id parentmeshA_parentmeshB. A mesh id 733_6
means, mesh id 733 with some flip denoted as 6. Make sure the
parentmeshA_parentmeshB pair is disjoint in train and validation.
